samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -0,0 +1,147 @@
+from botocore.exceptions import ClientError, ParamValidationError
+from botocore.response import StreamingBody
+
+from samcli.lib.remote_invoke.exceptions import InvalideBotoResponseException, InvalidResourceBotoParameterException
+from samcli.lib.remote_invoke.remote_invoke_executors import (
+    BotoActionExecutor,
+    RemoteInvokeExecutionInfo,
+    RemoteInvokeOutputFormat,
+    RemoteInvokeRequestResponseMapper,
+)
+from samcli.lib.utils import boto_utils
+
+LOG = logging.getLogger(__name__)
+FUNCTION_NAME = "FunctionName"
+PAYLOAD = "Payload"
+
+
+class LambdaInvokeExecutor(BotoActionExecutor):
+    """
+    Calls "invoke" method of "lambda" service with given input.
+    If a file location provided, the file handle will be passed as Payload object
+    """
+
+    _lambda_client: Any
+    _function_name: str
+
+    def __init__(self, lambda_client: Any, function_name: str):
+        self._lambda_client = lambda_client
+        self._function_name = function_name
+        self.request_parameters = {"InvocationType": "RequestResponse", "LogType": "Tail"}
+
+    def validate_action_parameters(self, parameters: dict) -> None:
+        """
+        Validates the input boto parameters and prepares the parameters for calling the API.
+
+        :param parameters: Boto parameters provided as input
+        """
+        for parameter_key, parameter_value in parameters.items():
+            if parameter_key == FUNCTION_NAME:
+                LOG.warning("FunctionName is defined using the value provided for --resource-id option.")
+            elif parameter_key == PAYLOAD:
+                LOG.warning(
+                    "Payload is defined using the value provided for either --payload or --payload-file options."
+                )
+            else:
+                self.request_parameters[parameter_key] = parameter_value
+
+    def _execute_action(self, payload: str):
+        self.request_parameters[FUNCTION_NAME] = self._function_name
+        self.request_parameters[PAYLOAD] = payload
+        LOG.debug(
+            "Calling lambda_client.invoke with FunctionName:%s, Payload:%s, parameters:%s",
+            self._function_name,
+            payload,
+            self.request_parameters,
+        )
+        try:
+            response = self._lambda_client.invoke(**self.request_parameters)
+        except ParamValidationError as param_val_ex:
+            raise InvalidResourceBotoParameterException(
+                f"Invalid parameter key provided."
+                f" {str(param_val_ex).replace('{FUNCTION_NAME}, ', '').replace('{PAYLOAD}, ', '')}"
+            ) from param_val_ex
+        except ClientError as client_ex:
+            if boto_utils.get_client_error_code(client_ex) == "ValidationException":
+                raise InvalidResourceBotoParameterException(
+                    f"Invalid parameter value provided. {str(client_ex).replace('(ValidationException) ', '')}"
+                ) from client_ex
+            raise client_ex
```

Comment:
I would urge that we re-catch and throw all exceptions as `click` exceptions. Maybe we are already doing that, but leaving it as a note in any case. :)

Reply:
Yup thats the plan. These exceptions would be handled in the context.py or command.py files.